util/project_allocation_mapper.py

```diff
@@ -341,6 +343,88 @@ def get_project_members(self, tas_project):
                 users.append(user)
         return users
 
+    def email_exists_on_project(self, project_id, email_address):
```

Comment:
Hmm, OK, I see that this one is harder to extract b/c it depends on other mapper functions. We can leave this in here and address it later as part of a larger refactor.